site/docs/introduction/aggregate-programming.md

```diff
@@ -0,0 +1,84 @@
+
+1. **Functions**: Expressed as `b(e1, ..., en)`, these apply a function `b` to arguments `e1, ..., en`. These functions include mathematical, logical, and algorithmic built-ins, as well as user-defined functions, and may represent sensors or actuators.
+
+2. **State variables (`rep`)**: The construct `rep(x ← v) s1; ...; sn` introduces a local state variable `x`, initialized with `v` and periodically updated by executing the block `s1; ...; sn`. This enables the definition of fields that dynamically evolve over time.
+
+3. **Neighbor values (`nbr`)**: The operation `nbr(s)` collects the latest values from all neighboring devices (including the device itself). Functions like `minHood(m)` synthesize these values, for example, by determining the minimum value in the field `m`.
+
+4. **Conditional branching (`if-else`)**: The construct `if(e) s1; ...; sn else s1'; ...; sm'` divides the network into two regions. In regions where the expression `e` evaluates as true, `s1; ...; sn` is executed, while `s1'; ...; sm'` is executed in other regions. Importantly, the branches remain separate and do not interact.
+
+### Resilient coordination operators
+
+The next level of abstraction within the aggregate programming framework introduces resilience and identifies a set of general basic operators designed for use in resilient coordination applications. This layer, positioned at the center in figure, includes coordination mechanisms that exhibit self-stabilization properties, i.e., the ability to adapt reactively to changes in the network structure or input values.
+
+A possible collection of basic operators includes three generalized coordination operators, along with "if" and built-in functions. The three operators are:
+
+1. **G(source, init, metric, accumulate)**: This operation represents a "spreading" operation that generalizes distance measurement, broadcasting, and projection operations. It performs two main tasks: first, it calculates a field of minimum distances from a specified source region (represented by a boolean field, `source`) using a provided metric. Then, it propagates values along the resulting distance gradient, starting from an initial value (`init`) and accumulating further values along the gradient using an accumulation function (`accumulate`).
+
+2. **C(potential, accumulate, local, null)**: This operation is responsible for accumulating information towards the source along the gradient of a potential field. Starting from an idempotent neutral element (`null`), the local value is combined with the "uphill" values using a commutative and associative accumulation function (`accumulate`), resulting in a cumulative value at the source.
+
+3. **T(initial, floor, decay)**: This operation describes a flexible countdown process characterized by a rate that may change over time. The "decay" function progressively reduces the initial value (`initial`) until it reaches the minimum allowed value (`floor`).
+
```

Comment:
same as I said for the constructs
In Collektive we don't call the functions G, C etc, just describe what they do and eventually cite the reference papers.